content/knowledge/consensus_algorithms.md

```diff
@@ -0,0 +1,367 @@
+
+When a client creates a transaction, it names one or more parent transactions, which are included inseparably in the inputs and their connections form the edges of the DAG.
+This parent-child relationship encoded in the DAG may or may not correspond to application specific dependencies — for example a child transaction need not spend to have relationship with the funds received in parent transaction.
+Transactions that are reachable via parent edges back in history are called _ancestor set_ and children transactions and their offspring are called _progeny_.
+
+The central challenge in the maintenance of the DAG is to choose among conflicting transactions.
+The notion of conflict is application-defined and transitive and forms an equivalence relation.
+For example transactions that spend the same funds form a conflict set and only one of them can be accepted.
+
+Avalanche instantiates a [Snowball](https://medium.com/@zaver.max/exploring-liveness-of-avalanche-d22f13b2db00) instance for each conflict set.
+Snowball uses repeated queries and counters to capture the amount of confidence built in conflicting transactions.
+
+In a given round each validator randomly selects a predefined number `n` of nodes from its validator list to query for their preferred decision on a value.
+Each queried validator responds with their preferred decisions, and if the majority of responses returned in a round differ from the one the node performing the query has, then it will update its own preferred decision to reflect that and respond to other nodes with that answer.
+All other validators process the same steps with subsamples of their validator lists.
+
+When a transaction is queried, all transactions reachable from it by following the DAG edges towards its ancestor set and its progeny are implicitly part of the query.
+A node will respond positively to the query if the transaction and its entire ancestry are currently the preferred option in their respective conflict sets.
+If a specific threshold of positive responders is reached, the transaction is set to collect a _chit_.
+Nodes compute confidence in transactions as the total amount of chits they obtained - they query a transaction only once and rely on new vertices and possible chits added to the progeny to build up their confidence.
+
+TODO: add a picture of avalanche protocol steps
+
+Nodes perform the querying round independently and asynchronously.
+So since one node does not query every other node like in classical consensus, each node performs their own sample of randomly selected nodes, thus able to scale to hundreds of thousands or millions of nodes without adding lots of additional overhead on each node, as they are querying the same number of nodes.
+
+With enough confidence, a decision is finalized immediately.
+This process happens so quickly that Avalanche rivals major payment systems in its ability process and clear transactions in a network.
```

Comment:
Let's remove the marketing talk. It happens quickly enough to have a high throughput of payments on its network.

Reply:
Removed, thanks!